File: p002/p2.cpp
#include <iostream>
using namespace std;

int main() {
	int sum = 0;
	int fib = 1, a = 1, b = 1;
	do {
		if (fib % 2 == 0)
			sum += fib;

		fib = a + b;
		b = a;
		a = fib;
	} while (fib < 4000000);

	cout << sum << endl;
	return 0;
}

File: p002/p2.rs
fn main() {
	let mut sum = 0;
	let mut fib = 1;
	let mut a = 1;
	let mut b = 1;
	
	while fib < 4_000_000 {
		if fib % 2 == 0 {
			sum += fib;
		}
		
		fib = a + b;
		b = a;
		a = fib;
	}
	println!("{}", sum);
}
